tdp/core/dag.py

```diff
@@ -145,14 +145,17 @@ def custom_key(node):
             )
             return f"{operation_priority:02d}_{node}"
 
-        def to_restart(node):
+        def to_node(node):
             if restart:
-                return node.replace("_start", "_restart")
-            return node
+                node_replaced = node.replace("_start", "_restart")
+                node_restart = self.collections.operations.get(node_replaced)
+                if node_restart is not None:
```

Comment:
I do not understand why you add this condition ? If the `restart` action does not exit the code must raise an exception. If I understand the new code, the `start` action will be used instead of the `restart` if the `restart` does not exists.

Reply:
Noop will also go through this part, and noop starts don't have a restart playbook (since they don't even have a start playbook)

Reply:
We can get the actual operation with `self.collections.operations.get(node)` and check if it is a `noop` ? If the operation is `noop` we do not perform the `node.replace("_start", "_restart")`. I think we must keep the error when a `restart` action is missing because without `restart` the new configuration will not be applied.

Reply:
If start operation was a noop, a noop restart operation will be created instead of returning the original start operation, since the caller expects to only get restart operation as a result

Reply:
ok